packages/mermaid/src/diagrams/class/classTypes.ts

```diff
@@ -3,13 +3,13 @@ import { parseGenericTypes, sanitizeText } from '../common/common.js';
 
 export interface ClassNode {
   id: string;
-  type: string;
-  label: string;
+  type?: string;
+  label?: string;
   shape: string;
-  text: string;
+  text?: string;
   cssClasses: string;
   methods: ClassMember[];
-  members: ClassMember[];
+  attributes: ClassMember[];
```

Comment:
This change needs to be documented in a changeset. I'm not really sure if we should consider this as breaking or not, but it should atleast be a minor version bump. @aloisklink wdyt?

Reply:
I don't see why it should be documented in a changeset.

Are we exposing the class types externally to any users of Mermaid?

The only thing I can exported that is potentially risky is the `InternalHelpers` type: https://github.com/mermaid-js/mermaid/blob/037ba2fa9cf40905bd1c35b1a450c5218a698f0f/packages/mermaid/src/internals.ts#L33, which looks to be used by the `@mermaid-js/layout-elk` package:

https://github.com/mermaid-js/mermaid/blob/037ba2fa9cf40905bd1c35b1a450c5218a698f0f/packages/mermaid/src/internals.ts#L15-L33

But it doesn't look like this `ClassNode` is being exported anywhere.

Reply:
Although we are not exporting it, it's being used by others like excalidraw. So it's good to be documented in the changelog.

Reply:
Ah, we are exporting it, just not via the `import XXX from 'mermaid';` entry-point.

We should probably ban `import type XXX from 'mermaid/dist/xxxxx'` by removing this line in our `package.json`: https://github.com/mermaid-js/mermaid/blob/d4e5acc530742c6bab3971cfe135137c774ae613/packages/mermaid/package.json#L14 (although this will be a breaking change), as otherwise every future change in the internal types will be a breaking change.

Maybe we can create a separate entry-point like `import type XXX from 'mermaid/internal-types'` where all these types are explicitly marked as not covered by SemVer and may change even between patch releases?

Or we could use something like <https://api-extractor.com/> to mark types as `@alpha`/`@beta`/`@internal` and warn us about breaking changes to the types automatically.

Reply:
Should it? Agree with Alois that internal names should not be an issue that causes a version bump.

Reply:
> This change needs to be documented in a changeset. I'm not really sure if we should consider this as breaking or not, but it should atleast be a minor version bump.

@sidharthv96 I was trying to create the changeset and running into an error.  Can you look into this for me?  I have not had power for 3 days, and current estimate is ~Monday (ack!), so will probably not be able to follow up on this again until then.

```
PS C:\source\repos\mermaid> pnpm changeset
🦋  error Error: Failed to find where HEAD diverged from master. Does master exist?
🦋  error     at getDivergedCommit (C:\source\repos\mermaid\node_modules\.pnpm\@changesets+git@3.0.1\node_modules\@changesets\git\dist\changesets-git.cjs.js:66:11)
🦋  error     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
🦋  error     at async getChangedFilesSince (C:\source\repos\mermaid\node_modules\.pnpm\@changesets+git@3.0.1\node_modules\@changesets\git\dist\changesets-git.cjs.js:198:22)
🦋  error     at async Object.getChangedPackagesSinceRef (C:\source\repos\mermaid\node_modules\.pnpm\@changesets+git@3.0.1\node_modules\@changesets\git\dist\changesets-git.cjs.js:239:24)
🦋  error     at async getVersionableChangedPackages (C:\source\repos\mermaid\node_modules\.pnpm\@changesets+cli@2.27.9\node_modules\@changesets\cli\dist\changesets-cli.cjs.js:245:27)
🦋  error     at async add (C:\source\repos\mermaid\node_modules\.pnpm\@changesets+cli@2.27.9\node_modules\@changesets\cli\dist\changesets-cli.cjs.js:509:35)
🦋  error     at async run (C:\source\repos\mermaid\node_modules\.pnpm\@changesets+cli@2.27.9\node_modules\@changesets\cli\dist\changesets-cli.cjs.js:1345:5)
```